### PR TITLE
add test for loading reporting settings

### PR DESCRIPTION
**Pull request by malaffoon**
Mostly demonstrating the way to test loading from yaml files.

### CODE REVIEW

File: common/src/test/resources/application-test-reporting-settings.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    EconomicDisadvantage: disabled
+    EnglishLanguageAcquisitionStatus: admin
+    Ethnicity: enabled
+    Gender: enabled
+    IndividualEducationPlan: admin
+    LimitedEnglishProficiencies: disabled
+    MigrantStatus: admin
+    MilitaryStudentIdentifier: admin
+    PrimaryLanguage: admin
+    Section504: admin
```
Comment (malaffoon) [inline common/src/test/resources/application-test-reporting-settings.yml]: Landon, is there anything else you want me to test as long as i have this going?

File: common/src/test/resources/application-test-reporting-settings.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+reporting:
+  state:
+    code: CA
+  iris-vendor-id: bob
+  studentFilters:
```
Comment (malaffoon) [inline common/src/test/resources/application-test-reporting-settings.yml]: One thought (and i apologize in advance): these settings are going to be used to suppress data in exports as well as suppress the filters. So, is `studentFilters` a bit too specific? Should it be `studentFields` or some such?

File: common/src/test/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettingsTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        assertThat(studentFilters.get(StudentFilterType.EnglishLanguageAcquisitionStatus)).isEqualTo(Admin);
+        assertThat(studentFilters.get(StudentFilterType.Ethnicity)).isEqualTo(Enabled);
+        assertThat(studentFilters.get(StudentFilterType.Gender)).isEqualTo(Enabled);
+        assertThat(studentFilters.get(StudentFilterType.IndividualEducationPlan)).isEqualTo(Admin);
+        assertThat(studentFilters.get(StudentFilterType.LimitedEnglishProficiencies)).isEqualTo(Disabled);
+        assertThat(studentFilters.get(StudentFilterType.MigrantStatus)).isEqualTo(Admin);
+        assertThat(studentFilters.get(StudentFilterType.MilitaryStudentIdentifier)).isEqualTo(Admin);
+        assertThat(studentFilters.get(StudentFilterType.PrimaryLanguage)).isEqualTo(Admin);
+        assertThat(studentFilters.get(StudentFilterType.Section504)).isEqualTo(Admin);
+        assertThat(studentFilters).hasSize(10);
```
Comment (esotrope) [inline common/src/test/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettingsTest.java]: why not use the recursive field by field compare as it covers size etc.

File: common/src/test/resources/application-test-reporting-settings.yml (hunk 2 above)
Comment (esotrope) [reply common/src/test/resources/application-test-reporting-settings.yml]: Interesting point. i think that makes more sense to call it `studentFields` if that is the case

File: common/src/test/resources/application-test-reporting-settings.yml (hunk 1 above)
Comment (esotrope) [reply common/src/test/resources/application-test-reporting-settings.yml]: you should also test omission of an entry maybe. other than that i think we're good

File: common/src/test/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettingsTest.java (hunk 3 above)
Comment (malaffoon) [reply common/src/test/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettingsTest.java]: Mostly cause there isn't an easy constructor for a map with more than 4 entries.

File: common/src/test/resources/application-test-reporting-settings.yml (hunk 2 above)
Comment (malaffoon) [reply common/src/test/resources/application-test-reporting-settings.yml]: Okay, will rename and update PR.

File: common/src/test/resources/application-test-reporting-settings.yml (hunk 1 above)
Comment (malaffoon) [reply common/src/test/resources/application-test-reporting-settings.yml]: Will do.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/SearchFieldPermissionLevel.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.opentestsystem.rdw.reporting.common.model;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/SearchFieldPermissionLevel.java]: just a rename, i didn't add/delete the files

**Pull request merged**